datasets/il/wmd_banned/crawler.py

```diff
@@ -0,0 +1,244 @@
+from zavod import Context
+from typing import List, Optional, Tuple
+from rigour.mime.types import XLSX
+import openpyxl
+from zavod import helpers as h
+import shutil
+import re
+import datetime
+
+
+# Hebrew to Gregorian month mapping
+HEBRW_MONTHS = {
+    "ינואר": "January",
+    "פברואר": "February",
+    "מרץ": "March",
+    "אפריל": "April",
+    "מאי": "May",
+    "יוני": "June",
+    "יולי": "July",
+    "אוגוסט": "August",
+    "ספטמבר": "September",
+    "אוקטובר": "October",
+    "נובמבר": "November",
+    "דצמבר": "December",
+}
+
+
+def hebrew_to_datetime(date_str: str):
+    """
+    Convert a Hebrew date string to a Normal datetime.date object.
+    """
+    if not date_str:
+        return None
+
+    if isinstance(date_str, int):
+        return datetime.date(date_str, 1, 1)
+
+    date_str = re.sub("\n", "", date_str)
+    for heb_month, eng_month in HEBRW_MONTHS.items():
+        date_str = re.sub(heb_month, f"{eng_month} ", date_str, flags=re.IGNORECASE)
+        date_str = re.sub("ב", "", date_str, flags=re.IGNORECASE)
+
+    date_fmt = ["%d %B %Y", "%d %B. %Y"]
+    for fmt in date_fmt:
+        try:
+            date_obj = datetime.datetime.strptime(date_str, fmt)
+            return date_obj.date()
+        except ValueError:
+            pass
+
+
+def extract_passport_no(text: str):
+    """
+    Extract passport numbers from a given text.
+    """
+    if not text:
+        return None
+    text = str(text)
+    pattern = r"\b[A-Z0-9]{5,}\b"
+    matches = re.findall(pattern, text)
+
+    return matches
+
+
+def extract_n_pop_address(text: str):
+    """
+    Extract address and update the text by removing the extracted address.
+    """
+    if not text:
+        return None, None
+
+    pattern = r"(address|location):\s*(.*)"
+    match = re.search(pattern, text, re.DOTALL | re.IGNORECASE)
+    if match:
+        address = match.group(2).strip()
+
+        # Remove the extracted location from the original text
+        updated_text = re.sub(
+            pattern, "", text, flags=re.DOTALL | re.IGNORECASE
+        ).strip()
+
+        return address, updated_text
+
+    else:
+        return None, text
+
+
+def clean_name(name: str | List[str]):
+    """
+    Clean the given name (or list of names) by removing newlines, quotes, colons, and trailing dots
+    """
+    if isinstance(name, list):
+        return [re.sub(r"[\n':]|^\.*|\.*$", "", match).strip() for match in name]
+
+    return re.sub(r"[\n':]|^\.*|\.*$", "", name).strip()
```

Comment:
do we want to remove these, or split on them?
I suspect h.multi_split(names_str, "1:", "2:", ..., ";", "\n", "ידוע גם:"]) would give us a lovely list of names.

Reply:
Possibly that removes the need for format_numbered_listing?

Reply:
ahh ok, makes sense